src/app/config/or_state_st.h

```diff
@@ -87,6 +87,18 @@ struct or_state_t {
 
   /** When did we last rotate our onion key?  "0" for 'no idea'. */
   time_t LastRotatedOnionKey;
+
+  /** Do we automatically fail IPv4 or IPv6? If so, set one of these to 1. */
+  int IPv4AutoFail;
+  int IPv6AutoFail;
+
+  /** When did we fail IPv4 or IPv6? */
+  time_t IPv4AutoFailTime;
+  time_t IPv6AutoFailTime;
+
+  /** Count the number of failures for IPv4 and IPv6 respectively. */
+  int IPv4Fails;
```

Comment:
The state file variables are unsigned int, but the struct is int. Use uint32_t for both.

Reply:
I opened a ticket for automatic checks that could avoid these kinds of issues in future:
https://trac.torproject.org/projects/tor/ticket/29115#ticket